### DISCUSSION TITLE
The fastest design given the fpga platform

### DISCUSSION THREAD

**Discussion by dj-park** [Xilinx/finn#838] [Q&A] [answered]
https://github.com/Xilinx/finn/blob/main/notebooks/end2end_example/cybersecurity/3-build-accelerator-with-finn.ipynb

From the link above, I want to have the fastest design given the fpga part (let's say xc7z020clg400-1 as used in the example above).
So, I set like below:

```
cfg_estimates = build.DataflowBuildConfig(
    output_dir          = estimates_output_dir,
    mvau_wwidth_max     = 10000,
    target_fps          = 100000000,
    synth_clk_period_ns = 10.0,
    fpga_part           = "xc7z020clg400-1",
    steps               = build_cfg.estimate_only_dataflow_steps,
    generate_outputs=[
        build_cfg.DataflowOutputType.ESTIMATE_REPORTS,
    ]
)
```
Note that I set mvau_wwidth_max and target_fps value to large values.

In the "**Launch a Build: Only Estimate Reports**" section, it says
"This particular report is useful to determine whether the current configuration will fit into a particular FPGA. If you see that the resource requirements are too high for the FPGA you had in mind, you should consider lowering the target_fps."

Does this mean that it is user's responsibility to make sure that the output design is mappable on the specific FPGA part?
So maybe the strategy would be like 
1) set mvau_wwidth_max and target_fps to large values that can't be achieved
2) keep adding more parallelism to layers until the sum of resource estimates for all layers reaches to my FPGA part?

**Comment by auphelia** [comment, marked as answer]
Hi @dj-park ,

Thanks for your question and also suggestions on how to improve the automated folding!
Yes, currently it is the user's responsibility to make sure the design will fit on the specific FPGA part. There is an open PR with an experimental setup for an automated folding like you describe it, check it out here: https://github.com/Xilinx/finn/pull/438
